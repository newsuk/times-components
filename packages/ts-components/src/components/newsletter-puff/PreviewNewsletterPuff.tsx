import {
  InpContainer,
  InpCopy,
  InpSignupContainer,
  InpSignupCTAContainer,
  InpSignupHeadline,
} from './styles';
import { NewsletterPuffButton } from './NewsletterPuffButton';
import React from 'react';

type PreviewNewsletterPuffProps = {
  copy: string;
  headline: string;
};

export const PreviewNewsletterPuff = ({
  copy,
  headline,
}: PreviewNewsletterPuffProps) => (
  <InpContainer>
    <InpSignupContainer>
      <InpSignupHeadline>{headline}</InpSignupHeadline>
      <InpCopy>{copy}</InpCopy>
      <InpSignupCTAContainer>
        <NewsletterPuffButton style='button'/>
      </InpSignupCTAContainer>
    </InpSignupContainer>
  </InpContainer>
);
