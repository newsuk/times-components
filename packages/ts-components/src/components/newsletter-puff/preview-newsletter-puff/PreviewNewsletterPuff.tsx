import { InpContainer } from '../styles';

import {
  InpCopy,
  InpSignupContainer,
  InpSignupHeadline,
  InpSignupCTAContainer
} from '../newsletter/styles';

import { NewsletterPuffButton } from '../newsletter-puff-button/NewsletterPuffButton';
import React from 'react';

type PreviewNewsletterPuffProps = {
  copy: string;
  headline: string;
  section: string;
};

export const PreviewNewsletterPuff = ({
  copy,
  headline,
  section
}: PreviewNewsletterPuffProps) => (
  <InpContainer section={section}>
    <InpSignupContainer>
      <InpCopy>
        <InpSignupHeadline>{headline} </InpSignupHeadline>
        {copy}
        <InpSignupCTAContainer childStyle="link">
          <NewsletterPuffButton style="link" />
        </InpSignupCTAContainer>
      </InpCopy>
      <InpSignupCTAContainer childStyle="button">
        <NewsletterPuffButton style="button" />
      </InpSignupCTAContainer>
    </InpSignupContainer>
  </InpContainer>
);
