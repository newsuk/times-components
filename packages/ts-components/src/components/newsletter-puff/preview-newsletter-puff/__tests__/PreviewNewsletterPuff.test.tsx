import React from 'react';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { PreviewNewsletterPuff } from '../PreviewNewsletterPuff';

const renderComponent = () =>
  render(
    <PreviewNewsletterPuff
      {...{
        code: 'TNL-119',
        section: 'news',
        headline: 'Politics. Explained.',
        copy:
          'Sign up to receive our brilliant Red Box newsletter, Matt Chorley`s poke at politics delivered every weekday morning at 8am.'
      }}
    />
  );

describe('Preview Newsletter Puff', () => {
  it('renders', () => {
    const component = renderComponent();
    expect(component.baseElement).toMatchSnapshot();
  });
});
