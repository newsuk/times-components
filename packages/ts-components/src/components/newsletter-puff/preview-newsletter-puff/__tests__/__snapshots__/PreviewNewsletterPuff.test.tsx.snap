// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Newsletter Puff renders 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa sc-htpNat ljXtVU"
    >
      <div
        class="sc-bdVaJa sc-EHOje hmOAGx"
      >
        <div
          class="sc-bwzfXH sc-gzVnrw keqsZo"
        >
          <div
            class="sc-bwzfXH sc-bZQynM khoGuQ"
          >
            Politics. Explained.
             
          </div>
          Sign up to receive our brilliant Red Box newsletter, Matt Chorley\`s poke at politics delivered every weekday morning at 8am.
          <div
            class="sc-bdVaJa sc-htoDjs lnzLgr"
          >
            <button
              class="sc-gZMcBi jWMqgP"
            >
              Sign up with one click
            </button>
          </div>
        </div>
        <div
          class="sc-bdVaJa sc-htoDjs fqXCoz"
        >
          <button
            class="sc-iwsKbI bmNDTm"
          >
            Sign up with one click
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
